### PR TITLE
Vagrantfile starts an ansible provisioner

### PR DESCRIPTION
**Pull request by m88614**
Super simple example of workflow starting nomad-jobs on the vagrant-box. 

Nikita is looking into problems with hive (and so am I), but I thought we might as well have this up for now as a working example.

### CODE REVIEW

File: example/resources/ansible/setup.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        name: nomad
+        state: restarted
+    - name: wait for consul to come up
+      uri:
+        url: "http://localhost:8500/v1/catalog/nodes"
+        status_code: 200
+      register: result
+      until: result.status == 200
+      retries: 60
+      delay: 1
```
Comment (zhenik) [inline example/resources/ansible/setup.yml:20]: I think 1 sec delay before first request, does not make sense :) IMHO 30 secs at least.

File: example/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  # Hashicorp consul ui
+  config.vm.network "forwarded_port", guest: 8500, host: 8500, host_ip: "127.0.0.1"
+
+  # Hashicorp nomad ui
+  config.vm.network "forwarded_port", guest: 4646, host: 4646, host_ip: "127.0.0.1"
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.linked_clone = true
+    vb.memory = 2048
+  end
```
Comment (zhenik) [inline example/Vagrantfile:14]: I think we need to discuss best which are best options here. @fredrikhgrelland

Comment (fredrikhgrelland) [reply example/Vagrantfile:14]: That depends on how much we need. I guess 1024 is minimum for hive and metastore? Lets try to make it as slim as posible

File: example/resources/ansible/setup.yml (hunk 1 above)
Comment (fredrikhgrelland) [reply example/resources/ansible/setup.yml:20]: I think this is delay between retires. 3 secs is better in that case. A total of 3 minutes.

Comment (m88614) [reply example/resources/ansible/setup.yml:20]: In my experience both services become available in a matter of seconds, so why do you mean 3 sec is better, Fredrik?

File: example/Vagrantfile (hunk 2 above)
Comment (m88614) [reply example/Vagrantfile:14]: Should we set it to 512 considering it's countdash, or just leave it as is?

Comment (fredrikhgrelland) [reply example/Vagrantfile:14]: Does that work?

File: example/resources/ansible/setup.yml (hunk 1 above)
Comment (fredrikhgrelland) [reply example/resources/ansible/setup.yml:20]: [Thundering herd](https://en.wikipedia.org/wiki/Thundering_herd_problem#:~:text=In%20computer%20science%2C%20the%20thundering,able%20to%20handle%20the%20event.)

File: example/Vagrantfile (hunk 2 above)
Comment (m88614) [reply example/Vagrantfile:14]: Nope. 1024 does.

File: example/Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  config.vm.network "forwarded_port", guest: 8500, host: 8500, host_ip: "127.0.0.1"
+
+  # Hashicorp nomad ui
+  config.vm.network "forwarded_port", guest: 4646, host: 4646, host_ip: "127.0.0.1"
+
+  config.vm.provider "virtualbox" do |vb|
+    vb.linked_clone = true
+    vb.memory = 1024
+  end
+  config.vm.provision "ansible" do |ansible|
```
Comment (zhenik) [inline example/Vagrantfile:15]: @fredrikhgrelland mentioned `ansible_local`

Comment (m88614) [reply example/Vagrantfile]: Done!

**Pull request merged**